integration-tests/bats/fetch.bats

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+    TMPDIRS=$(pwd)/tmpdirs
+    mkdir -p $TMPDIRS/{rem1,repo1}
+
+    # repo1 -> rem1 -> repo2
+    cd $TMPDIRS/repo1
+    dolt init
+    dolt remote add origin file://../rem1
+    dolt remote add test-remote file://../rem1
+    dolt push origin main
+
+    cd $TMPDIRS
+    dolt clone file://rem1 repo2
+    cd $TMPDIRS/repo2
+    dolt branch feature
+    dolt remote add test-remote file://../rem1
+
+    # table and commits only present on repo1, rem1 at start
+    cd $TMPDIRS/repo1
+    dolt sql -q "create table t1 (a int primary key, b int)"
+    dolt add .
+    dolt commit -am "First commit"
+    dolt sql -q "insert into t1 values (0,0)"
+    dolt commit -am "Second commit"
+    dolt branch feature
+    dolt push origin main
+    cd $TMPDIRS
+}
+
+teardown() {
+    teardown_common
+    rm -rf $TMPDIRS
+    cd $BATS_TMPDIR
+}
+
+@test "fetch: basic fetch" {
+    cd repo2
+    dolt fetch
+
+    run dolt diff main origin/main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('origin/main')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
+}
+
+@test "fetch: fetch origin" {
+    cd repo2
+    dolt fetch origin
+
+    run dolt diff main origin/main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('origin/main')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
+}
+
+@test "fetch: fetch main" {
```

Comment:
This test isn't valid is there isn't a check for the absence of something. You should create another branch on the remote, and verify that it doesn't get fetched when you just fetch main.